core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -69,16 +74,35 @@ public FastAppend appendFile(DataFile file) {
     return this;
   }
 
+  @Override
+  public FastAppend appendManifest(ManifestFile manifest) {
+    // the manifest must be rewritten with this update's snapshot ID
+    try (ManifestReader reader = ManifestReader.read(
+        ops.io().newInputFile(manifest.path()), ops.current()::spec)) {
+      appendManifests.add(ManifestWriter.copyAppendManifest(
```

Comment:
Nit: I would, probably, split this line and an equivalent one in `MergeAppend`. 

```
OutputFile newManifestPath = manifestPath(manifestCount.getAndIncrement());
appendManifests.add(ManifestWriter.copyAppendManifest(reader, newManifestPath, snapshotId(), summaryBuilder));
```

Reply:
Fixed.